### DISCUSSION TITLE
How to set the portal for a Tooltip

### DISCUSSION THREAD

**Discussion by jasikpark** [ariakit/ariakit#2814] [Questions]
I have a Tooltip that is rendered inside of a Dialog - currently because the Dialog has a z-index of `var(--z-index-modal)` or `100`, the tooltip ends up rendered behind the modal.

![image](https://github.com/ariakit/ariakit/assets/10626596/63277098-b3c7-4a41-b9b0-6218523e74fb)

Is there an option I can pass to the tooltip to have it mount inside the Dialog rather than to the `<body>` level Portal?

**Comment by diegohaz** [comment]
If it's an Ariakit Dialog, the tooltip portal should be rendered within the dialog portal.

But you can also disable the tooltip portal:

```jsx
<Tooltip portal={false}>
```

**Reply by jasikpark** [reply]
It is an Ariakit Dialog.. it doesn't seem that it is rendered within the dialog. maybe something I need to fix in the render={} where we customize the background of the modal?

**Reply by diegohaz** [reply]
When set to `portal={true}` (the default setting), the tooltip will render within the dialog *portal*, not the dialog itself. If you wish to render the tooltip within the dialog, simply use `portal={false}` on the tooltip.
